test_requests.py

```diff
@@ -1265,6 +1265,25 @@ def test_get_environ_proxies(self):
             'http://localhost.localdomain:5000/v1.0/') == {}
         assert get_environ_proxies('http://www.requests.com/') != {}
 
+    def test_guess_filename_when_int(self):
+        from requests.utils import guess_filename
+        assert None is guess_filename(1)
+
+    def test_guess_filename_when_filename_is_an_int(self):
+        from requests.utils import guess_filename
+        fake = type('Fake', (object,), {'name': 1})()
+        assert None is guess_filename(fake)
+
+    def test_guess_filename_with_file_like_obj(self):
+        from requests.utils import guess_filename
+        fake = type('Fake', (object,), {'name': 'value'})()
+        assert 'value' == guess_filename(fake)
+
+    def test_guess_filename_with_unicode_name(self):
+        from requests.utils import guess_filename
+        fake = type('Fake', (object,), {'name': b'value'.decode('utf-8')})()
+        assert 'value' == guess_filename(fake)
```

Comment:
I disagree. The test will work just fine on Py2. And we do want to check the value since we want to make sure the name is returned. At the very least we want to make sure it isn't `None`.